lib/ecto.ex

```diff
@@ -291,17 +289,27 @@ defmodule Testcontainers.Ecto do
               :ok
           end
 
-        Ecto.Migrator.run(repo, absolute_migrations_path, :up, all: true)
-
-        GenServer.stop(pid)
-
-        {:ok, container}
+        with {:ok, _, _} <- run_migrations(repo, absolute_migrations_path) do
+          {:ok, container}
+        end
 
       {:error, reason} ->
         {:error, reason}
     end
   end
 
+  defp run_migrations(repo, migrations_path) do
+    try do
+      Ecto.Migrator.with_repo(
+        repo,
+        &Ecto.Migrator.run(&1, migrations_path, :up, all: true)
+      )
```

Comment:
Please fix these warnings.

Reply:
the reason why these errors show up is because I test that its possible to build an example project that doesn't have ecto in dependencies. It should be possible to use testcontainers without ecto. On the other side I have made an issue here https://github.com/testcontainers/testcontainers-elixir/issues/53 previously about the need for (possibly?) to move the ecto logic into its own package, which then depends on testcontainers for elixir